From 98c398592cc1eb2d72388f1216235f1d61558732 Mon Sep 17 00:00:00 2001
From: Charles Stevens <cstevens@beaconembedded.com>
Date: Fri, 21 Jan 2022 13:31:47 -0600
Subject: [PATCH 68/68] arm64: renesas: beacon: Fix USB-C device mode

Add missing device_release_driver() in the NONE role case. This
should match up to the way it works in the manual role switch case.

Fixes: RSOM-360

Signed-off-by: Charles Stevens <cstevens@beaconembedded.com>
---
 drivers/usb/gadget/udc/renesas_usb3.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/udc/renesas_usb3.c b/drivers/usb/gadget/udc/renesas_usb3.c
index d9668f86a477..97071d90ed4f 100644
--- a/drivers/usb/gadget/udc/renesas_usb3.c
+++ b/drivers/usb/gadget/udc/renesas_usb3.c
@@ -2367,6 +2367,7 @@ static void handle_ext_role_switch_states(struct device *dev,
 		break;
 	case USB_ROLE_DEVICE:
 		if (usb3->connection_state == USB_ROLE_NONE) {
+			device_release_driver(host);
 			usb3->connection_state = USB_ROLE_DEVICE;
 			usb3_set_mode(usb3, false);
 			if (usb3->driver)
-- 
2.17.1

